Added tls settings for creating or updating hosts via api

dj-wasabi opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
zabbix_host

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible --version
ansible 2.2.0.0
  config file = /git/ansible/environment/ansible.cfg
  configured module search path = ['Library']
```

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Updated the `zabbix_host` module that also it also can configure some TLS settings that have been added with Zabbix 3.0 and later when creating or updating an host.

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

<!-- Paste verbatim command output below, e.g. before and after your change -->
```

```

Commit:
Added tls settings for creating or updating hosts via api

Commit:
Added tls settings for creating or updating hosts via api

bcoca commented on lib/ansible/modules/monitoring/zabbix_host.py:
```diff
@@ -119,6 +120,44 @@
+            - The tls_connect parameter accepts values of 1 to 7
+            - Possible values, 1 (no encryption), 2 (PSK), 4 (certificate).
+            - Values can be combined.
+        default: 1
```
it would be nicer to allow a list of 'text options':
``` tls_connect: none```
or
```tls_connect: PSK, certificate```

dj-wasabi replied:
Yes, I've made changes. tls_connect is a string, the tls_accept is now a list:
```
tls_accept:
  - PSK
  - certificate
```

Commit:
Created function for calculating encryption level

dj-wasabi commented:
CI fails on 1 sub job:
```
sudo docker pull drydock/u14pyt:prod 0s  
Warning: failed to get default registry endpoint from daemon (Cannot connect to the Docker daemon. Is the docker daemon running on this host?). Using system default: https://index.docker.io/v1/
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```
How should I proceed?

Commit:
Correcting name

frennkie commented:
Just pulled this PR into a separate branch and tested by setting the following:

```
  local_action:
    module: zabbix_host
    [...]
    tls_connect: "PSK"
    tls_accept: ["PSK"]
    tls_psk_identity: "{{ zabbix_psk_identity }}"
    tls_psk: "{{ zabbix_psk }}"
```

Works as expected.

mattclay commented:
Closing and re-opening to trigger CI.

Akasurde commented:
LGTM except rebase.

Commit:
Updating to 2.4

dj-wasabi commented:
ready_for_review

henryk committed:
Fix missing parameters for new interface

henryk committed:
Get rid of TLS parameters in link_or_clear_template() call. The Zabbix API states that only those parameters in host.update are changed that are present in the call. We don't need to send all parameters all the time. (Note: Need to properly check in in check_all_properties() though.)

henryk committed:
Remove `exist_interfaces_copy`. There doesn't seem to be a reason for it to exist: `exist_interfaces` is never changed.

henryk committed:
Update check_all_properties() to check for new properties and fix checks for the existing properties (proxy_hostid used to give false positives). Remove the case distinction in the main flow: There doesn't seem to be a useful difference (except that the `else` clause updates the inventory mode and the `if` clause doesn't) and logically it doesn't make sense to compare the lengths of the interface lists, since check_all_properties() is supposed to properly compare their contents including length.

henryk committed:
Make `interface_list_copy` a proper copy. There's some modification going on that I don't fully understand, but `interface_list_copy = exist_interface_list` would not create a copy even though the variable naming seems to imply that. Add a proper deepcopy call.

henryk committed:
Simplify check_interface_properties(). Both input lists are now treated equally (using idiomatic "augment, sort, compare") and converted into flat lists of values which can then be compared with the built-in operator.

Commit:
Merge pull request #1 from henryk/adding_tls_settings_zabbix_host

Adding tls settings zabbix host

Commit:
Fix merge issue

giovannisciortino committed:
Fixes #27635: pam_limits fix add comment (#27999)

dj-wasabi commented:
ready_for_review

alikins committed:
prompt for new pass on create/encrypt if none specified (#28185)

* prompt for new pass on create/encrypt if none specified

Make 'ansible-vault' edit or encrypt prompt for a password
if none or provided elsewhere.

Note: ansible-playbook does not prompt if not vault password
is provided

* dont show vault password prompts if not a tty

Commit:
fix for remove idempotent issue (#28226)

Commit:
Fix merge issue

dj-wasabi commented:
ready_for_review

Akasurde commented on lib/ansible/modules/monitoring/zabbix_host.py:
```diff
@@ -2,12 +2,26 @@
 
 # (c) 2013-2014, Epic Games, Inc.
-# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
```
Could you please revert this to one liner license?

bcoca replied:
this is the preference going forward

Akasurde commented on lib/ansible/modules/monitoring/zabbix_host.py:
```diff
@@ -442,7 +570,7 @@ def main():
 
     if not HAS_ZABBIX_API:
-        module.fail_json(msg="Missing required zabbix-api module (check docs or install with: pip install zabbix-api)")
+        module.fail_json(msg="Missing requried zabbix-api module (check docs or install with: pip install zabbix-api)")
```
Why change correct spelling to wrong one ?

Commit:
Fixing comments
